java/custom/com/nokia/mid/s40/bg/BGUtils.java

```diff
@@ -1,13 +1,52 @@
 package com.nokia.mid.s40.bg;
 
+import com.sun.midp.main.MIDletSuiteUtils;
+
 public class BGUtils {
+    private static boolean launchMIDletCalled = false;
+
+    private static native int getFGMIDletNumber();
+    private static native String getFGMIDletClass();
+    private static native void waitUserInteraction();
+
+    /* Start the FG MIDlet when the page becomes visible and if
+       launchIEMIDlet hasn't been called (we want launchIEMIDlet
+       to launch the MIDlet if possible) */
     public static void setBGMIDletResident(boolean param) {
-        System.out.println("warning: BGUtils.setBGMIDletResident(Z)V not implemented (" + param + ")");
+      BGUtils.waitUserInteraction();
+
+      if (BGUtils.launchMIDletCalled) {
+        return;
+      }
+
+      int midletNumber = BGUtils.getFGMIDletNumber();
+      String midletClass = BGUtils.getFGMIDletClass();
+
+      MIDletSuiteUtils.execute(midletNumber, midletClass, null);
     }
 
-    public static boolean launchIEMIDlet(String string1, String string2, int int1, String string3, String string4) {
-        throw new RuntimeException("BGUtils.launchIEMIdlet(L...String;L...String;IL...String;L...String;)Z " +
-                                   "not implemented (" + string1 + ", " + string2 + ", " + int1 + ", " + string3 +
-                                   ", " + string4 + ")");
+    private static native void addSystemProperties(String args);
+
+    public static boolean launchIEMIDlet(String midletSuiteVendor, String midletName, int midletNumber, String startupNoteText, String args) {
+        System.out.println("midletSuiteVendor: " + midletSuiteVendor);
+        System.out.println("midletName: " + midletName);
+        System.out.println("midletNumber: " + midletNumber);
+        System.out.println("startupNoteText: " + startupNoteText);
+        System.out.println("args: " + args);
```

Comment:
Yes, removing logging is included in the set of things I wanted to do in this PR (see my first comment)